packages/cli/src/ui/components/shared/text-buffer.test.ts

```diff
@@ -1521,6 +1521,75 @@ Contrary to popular belief, Lorem Ipsum is not simply random text. It has roots
     });
   });
 
+  describe('inputFilter', () => {
+    it('should filter input based on the provided filter function', () => {
+      const { result } = renderHook(() =>
+        useTextBuffer({
+          viewport,
+          isValidPath: () => false,
+          inputFilter: (text) => text.replace(/[^0-9]/g, ''),
```

Comment:
nit: it might make sense to also apply the inputFilter to the initial value passed in but not sure that is a good idea.